test/Microsoft.AspNetCore.Razor.Tools.Test/RequestDispatcherTest.cs

```diff
@@ -0,0 +1,552 @@
+                        var connectionTask = CreateConnectionWithEmptyBuildRequest(c =>
+                        {
+                            c.WaitForDisconnectAsyncFunc = _ => source.Task;
+                        });
+                        list.Add(source);
+                        return connectionTask;
+                    }
+
+                    readySource.SetResult(true);
+                    return new TaskCompletionSource<Connection>().Task;
+                });
+
+            var compilerHost = CreateCompilerHost(c =>
+            {
+                c.ExecuteFunc = (req, ct) =>
+                {
+                    return EmptyBuildResponse;
+                };
+            });
+
+            var keepAlive = TimeSpan.FromSeconds(1);
+            var eventBus = new TestableEventBus();
+            var dispatcherTask = Task.Run(() =>
+            {
+                var dispatcher = new DefaultRequestDispatcher(connectionHost.Object, compilerHost, CancellationToken.None, eventBus, keepAlive);
+                dispatcher.Run();
+            });
+
+            await readySource.Task;
+            foreach (var source in list)
+            {
+                source.SetResult(true);
+            }
+
+            // Act
+            await dispatcherTask;
+
+            // Assert
+            Assert.Equal(totalCount, eventBus.CompletedCount);
+            Assert.True(eventBus.LastProcessedTime.HasValue);
+            Assert.True(eventBus.HitKeepAliveTimeout);
+        }
+
+        [Fact]
+        public void ClientExceptionShouldBeginShutdown()
+        {
+            // Arrange
+            var listenCancellationToken = default(CancellationToken);
+            var firstConnectionTask = CreateConnectionWithEmptyBuildRequest(c =>
+            {
+                c.WaitForDisconnectAsyncFunc = (ct) =>
+                {
+                    listenCancellationToken = ct;
+                    return Task.Delay(Timeout.Infinite, ct).ContinueWith<Connection>(_ => null);
+                };
+            });
+            var secondConnectionTask = CreateConnectionWithEmptyBuildRequest(c =>
+            {
+                c.WaitForDisconnectAsyncFunc = (ct) => throw new Exception();
+            });
+
+            var compilerHost = CreateCompilerHost();
+            var connectionHost = CreateConnectionHost(
+                firstConnectionTask,
+                secondConnectionTask,
+                new TaskCompletionSource<Connection>().Task);
+            var keepAlive = TimeSpan.FromSeconds(10);
+            var eventBus = new TestableEventBus();
+            var dispatcher = new DefaultRequestDispatcher(connectionHost, compilerHost, CancellationToken.None, eventBus, keepAlive);
+
+            // Act
+            dispatcher.Run();
+
+            // Assert
+            Assert.True(eventBus.HasDetectedBadConnection);
+            Assert.True(listenCancellationToken.IsCancellationRequested);
+        }
+
```

Comment:
Is there a test for the happy path? Considering we have two commands, we should at least exercise the serialization functionality for both of those.

Reply:
You are right. Those tests will be there in an upcoming PR